src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -542,19 +545,40 @@ public static async Task<Authority> CreateAuthorityForRequestAsync(RequestContex
 
                     case AuthorityType.Aad:
 
-                        bool updateEnvironment = requestContext.ServiceBundle.Config.MultiCloudSupportEnabled && account != null && !PublicClientApplication.IsOperatingSystemAccount(account);
+                        bool isMultiCloudSupported = requestContext.ServiceBundle.Config.MultiCloudSupportEnabled && account != null && !PublicClientApplication.IsOperatingSystemAccount(account);
+
+                        bool useMtlsEnvironment = requestContext != null && requestContext.ShouldUseMtlsPop();
+
+                        bool updateEnvironment = isMultiCloudSupported || useMtlsEnvironment;
```

Comment:
This code can be simplified. You should no longer need updateEnvironment variable

Reply:
```
if (useMtlsEnvironment)
{
    // Use the mTLS specific authority creation
    return CreateAuthorityForMtlsWithTenant(
        configAuthorityInfo,
        account?.HomeAccountId?.TenantId);
}

if (isMultiCloudSupported)
{
    // Use the environment update with standard authority creation
    return CreateAuthorityWithTenant(
        CreateAuthorityWithEnvironment(configAuthorityInfo, account?.Environment),
        account?.HomeAccountId?.TenantId,
        forceSpecifiedTenant: false);
}

// Default non-mTLS environment authority creation
return CreateAuthorityWithTenant(
    configAuthority,
    account?.HomeAccountId?.TenantId,
    forceSpecifiedTenant: false);

Reply:
+1 - the cyclomatic complexity of this method is too high ... https://www.sonarsource.com/learn/cyclomatic-complexity/#:~:text=Cyclomatic%20complexity%20is%20a%20software,complexity%20metric%20becomes%20especially%20useful.